app/logic/users.py

```diff
@@ -10,28 +12,78 @@ def isEligibleForProgram(program, user):
     :param user: accepts a User object or userid
     :return: True if the user is not banned and meets the requirements, and False otherwise
     """
-
-    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program).exists()):
+    now = datetime.datetime.now()
+    if (ProgramBan.select().where(ProgramBan.user == user, ProgramBan.program == program, ProgramBan.endDate > now).exists()):
         return False
 
     return True
 
-def addRemoveInterest(rule, program_id, currentUser):
+def addUserInterest(program_id, username):
+    """
+    This function is used to add an interest to .
+    Parameters:
+    program_id: id of the program the user is interested in
+    username: username of the user showing interest
+    """
+    try:
+        Interest.get_or_create(program = program_id, user = username)
+        return "Successfully added interest"
+    except:
+        return "This interest does not exist"
+
+def removeUserInterest(program_id, username):
     """
     This function is used to add or remove interest from the interest table.
     Parameters:
-    rule: Gets the url from the ajax call, specifies to add or remove interest
     program_id: id of the program the user is interested in
+    username: username of the user showing disinterest
+
     """
+    try:
+        interestToDelete = Interest.get(Interest.program == program_id, Interest.user == username)
+        interestToDelete.delete_instance()
+        return "Successfully removed interest"
+    except:
+        return "This interest does not exist"
```

Comment:
If they want to remove an interest and one doesn't exist, who cares? The end result is the same. 

If this text doesn't ever show to the end user, I wouldn't worry about changing it, but a better plan would be to check for a `DoesNotExist` exception and ignore it. That way any other exception will still be raised for the controller to capture. The way you have it right now the try/except in the controller is unnecessary, because every exception is caught inside this function.

Reply:
Your change didn't actually address this, as far as I can tell.